### PR TITLE
[eth-rpc] Fixes

### PR DESCRIPTION
**Pull request by pgherveou**
Various fixes for the release of eth-rpc & ah-westend-runtime

- Bump asset-hub westend spec version
- Fix the status of the Receipt to properly report failed transactions
- Fix value conversion between native and eth decimal representation

### CODE REVIEW

**Review by athei — APPROVED** [review]

File: prdoc/pr_6317.prdoc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+title: Bump asset-hub-westend spec version
+doc:
+- audience: Runtime Dev
+  description: null
```
Comment (pgherveou) [inline prdoc/pr_6317.prdoc:4]:
```suggestion
  description: Bump asset-hub westend spec version
```

**Comment by pgherveou** [comment]
Re-purposed the PR to include various fixes noticed today during testing

File: cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs [hunk 2]
```diff
@@ -124,7 +124,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("westmint"),
 	impl_name: create_runtime_str!("westmint"),
 	authoring_version: 1,
-	spec_version: 1_016_002,
+	spec_version: 1_016_004,
```
Comment (pgherveou) [inline cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs:127]: note: bumping twice as 003 has been pushed to westend already

File: substrate/frame/revive/src/lib.rs [hunk 3]
```diff
@@ -303,6 +303,10 @@ pub mod pallet {
 		/// preventing replay attacks.
 		#[pallet::constant]
 		type ChainId: Get<u64>;
+
+		/// The ratio between the decimal representation of the native token and the ETH token.
+		#[pallet::constant]
+		type NativeToEthRatio: Get<u32>;
```
Comment (athei) [inline substrate/frame/revive/src/lib.rs:309]: Isn't handled by adding the network with the correct decimals to the wallet?

**Review by xermicus — APPROVED** [review]

File: substrate/frame/revive/src/lib.rs (hunk 3 above)
Comment (pgherveou) [reply substrate/frame/revive/src/lib.rs:309]: in theory yes, but in practice no wallet support anything else than 18 decimals https://github.com/MetaMask/metamask-extension/issues/15168

Comment (pgherveou) [reply substrate/frame/revive/src/lib.rs:309]: https://community.metamask.io/t/add-support-for-native-chain-tokens-not-18-decimals/25951

Comment (athei) [reply substrate/frame/revive/src/lib.rs:309]: Thanks!

**Pull request merged**